lib/simp/cli/commands/bootstrap.rb

```diff
@@ -1,66 +1,283 @@
+    end
+
+    bootstrap_start_time = Time.now
+
+    # Set us up to use the SIMP environment, if this has not already been done.
+    # (1) Be careful to preserve the existing primary, 'production' environment,
+    #     if one exists.
+    # (2) Create links to production in both the primary and secondary environment
+    #     paths.
+    environment_path = ::Utils.puppet_info[:simp_environment_path]
+    fail("Could not find the simp environment path at #{environment_path}") unless File.directory?(environment_path)
+
+    item = Simp::Cli::Config::Item::SetProductionToSimpAction.new
+    item.start_time = bootstrap_start_time
+    item.apply
+    fail ("Could not set 'simp' to production environment") unless item.applied_status == :succeeded
+
+    linecounts = Array.new
+
+    # Open log file
+    logfilepath = File.expand_path(BOOTSTRAP_LOG)
+    FileUtils.mkpath(File.dirname(logfilepath)) unless File.exists?(logfilepath)
+    @logfile = File.open(logfilepath, 'w')
+
+    # Print intro
+    system('clear')
+    say "=== Starting SIMP Bootstrap ===".yellow
+
+    # Set an interrupt trap if safe mode is enabled
+    say "> The log can be found at '#{@logfile.path}'\n"
+    if not @unsafe
+      signals = ["INT","HUP","USR1","USR2"]
+      signals.each do |sig|
+        Signal.trap(sig) { say 'Safe mode enabled, ignoring interrupt'.magenta }
+      end
+      say "> Interrupts will be captured and ignored to ensure bootstrap integrity.".magenta.bold
+    else
+      say "> WARNING: Any interrupts may cause system instability.".red.bold
+    end
+
+
+    # Kill all puppet processes and stop specific services
+    say "> Killing all Puppet processes".cyan
+    system("pkill -9 -f puppet >& /dev/null")
+    system('pkill -f pserver_tmp')
+    system("puppet resource service puppetserver ensure=stopped >& /dev/null")
+
+    # Kill the connection with puppetdb
+    say "> Killing connection to PuppetDB".cyan
+    system("puppet resource service puppetdb ensure=stopped >& /dev/null")
+    confdir = ::Utils.puppet_info[:config]['confdir']
+    if File.exists?("#{confdir}/routes.yaml")
+      system("rm -f #{confdir}/routes.yaml")
+      say "> DEBUG: Successfully removed #{confdir}/routes.yaml".green if @verbose
+    else
+      say "> DEBUG: Did not find #{confdir}/routes.yaml, not removing" if @verbose
+    end
+    system('puppet config set --section master storeconfigs false')
+    system('puppet config set --section main storeconfigs false')
+    say "> DEBUG: Successfully set storeconfigs=false in #{confdir}/puppet.conf".green if @verbose
+
+    # Figure out what to do with puppetserer certs
+    rm_certs = false
+    if @remove_certs then
+      rm_certs = true
+    # If remove_certs is not true OR false, it is nil and should be prompted for.
+    elsif not @remove_certs == false
+      rm_ask = ask("> Do you wish to keep existing puppetserver certificates? (yes|no) ") { |q| q.validate = /(yes)|(no)/i }
```

Comment:
Can you detect if there are certs here to avoid asking for user input?

There shouldn't be any before puppetserver is started for the first time (I think)